internal/configconverter/remove_memory_ballast_key_test.go

```diff
@@ -0,0 +1,64 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Taken from https://github.com/open-telemetry/opentelemetry-collector/blob/v0.66.0/confmap/converter/overwritepropertiesconverter/properties_test.go
+// to prevent breaking changes.
+package configconverter
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/confmap/confmaptest"
+)
+
+func TestRemoveMemoryBallastConverter_Empty(t *testing.T) {
+	pmp := RemoveMemoryBallastKey{}
+	conf := confmap.NewFromStringMap(map[string]interface{}{"foo": "bar"})
+	assert.NoError(t, pmp.Convert(context.Background(), conf))
+	assert.Equal(t, map[string]interface{}{"foo": "bar"}, conf.ToStringMap())
+}
+
+func TestRemoveMemoryBallastConverter_With_Memory_Ballast(t *testing.T) {
+	cfgMap, err := confmaptest.LoadConf("testdata/with_memory_ballast.yaml")
+	require.NoError(t, err)
+	require.NotNil(t, cfgMap)
+	pmp := RemoveMemoryBallastKey{}
+	assert.NoError(t, pmp.Convert(context.Background(), cfgMap))
+	assert.Equal(t, map[string]interface{}{"receivers": map[string]interface{}{"otlp": map[string]interface{}{"protocols": map[string]interface{}{"grpc": interface{}(nil), "http": interface{}(nil)}}}, "service": map[string]interface{}{"extensions": []interface{}{"health_check", "http_forwarder", "zpages", "smartagent"}, "pipelines": map[string]interface{}{"metrics": map[string]interface{}{"receivers": []interface{}{"otlp"}}}}}, cfgMap.ToStringMap())
+}
+
+func TestMemoryBallastConverter_Without_Memory_Ballast(t *testing.T) {
+	cfgMap, err := confmaptest.LoadConf("testdata/without_memory_ballast_config.yaml")
+	require.NoError(t, err)
+	require.NotNil(t, cfgMap)
+	pmp := RemoveMemoryBallastKey{}
+	assert.NoError(t, pmp.Convert(context.Background(), cfgMap))
+	assert.Equal(t, map[string]interface{}{"extensions": map[string]interface{}{"health_check": map[string]interface{}{"endpoint": "${SPLUNK_LISTEN_INTERFACE}:13133"}, "http_forwarder": map[string]interface{}{"egress": map[string]interface{}{"endpoint": "${SPLUNK_API_URL}"}, "ingress": map[string]interface{}{"endpoint": "${SPLUNK_LISTEN_INTERFACE}:6060"}}, "smartagent": map[string]interface{}{"bundleDir": "${SPLUNK_BUNDLE_DIR}", "collectd": map[string]interface{}{"configDir": "${SPLUNK_COLLECTD_DIR}"}}, "zpages": interface{}(nil)}, "service": map[string]interface{}{"extensions": []interface{}{"health_check", "http_forwarder", "zpages", "smartagent"}, "telemetry": map[string]interface{}{"metrics": map[string]interface{}{"address": "${SPLUNK_LISTEN_INTERFACE}:8888"}}}}, cfgMap.ToStringMap())
```

Comment:
It's a bit hard to read and maintain. Can you read `testdata/without_memory_ballast_config.yaml` again instead?

Reply:
I agree, read from the same configmap and also for the other test create an yaml file and reading it to create a expected map.